bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1190,6 +1190,20 @@ impl<'a> CircuitInputStateRef<'a> {
                 CallContextField::IsSuccess,
                 0u64.into(),
             );
+
+            //even call.rw_counter_end_of_reversion is zero for now, it will set in
+            //set_value_ops_call_context_rwc_eor later
+            // if call fails, no matter root or internal, read RwCounterEndOfReversion for
+            // circuit constraint.
+            if !call.is_success {
+                self.call_context_read(
+                    exec_step,
+                    call.call_id,
+                    CallContextField::RwCounterEndOfReversion,
+                    call.rw_counter_end_of_reversion.into(),
+                );
+            }
```

Comment:
The `if !call.is_success` here is unnecessary (see line 1185).

```suggestion
            // Even call.rw_counter_end_of_reversion is zero for now, it will set in
            // `set_value_ops_call_context_rwc_eor` later.
            // If call fails, no matter root or internal, read RwCounterEndOfReversion for
            // circuit constraint.
            self.call_context_read(
                exec_step,
                call.call_id,
                CallContextField::RwCounterEndOfReversion,
                call.rw_counter_end_of_reversion.into(),
            );
```

Reply:
thanks for catching it fixed in [faa3721](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1030/commits/faa372133d9079ff17be8f7346c059016dacb82a)